gst/nnstreamer/tensor_if/gsttensorif.c

```diff
@@ -231,145 +292,503 @@ static void
 gst_tensor_if_set_property (GObject * object, guint prop_id,
     const GValue * value, GParamSpec * pspec)
 {
-  /* GstTensorIf *self = GST_TENSOR_IF (object); */
-  /** @todo NYI! */
+  GstTensorIf *self = GST_TENSOR_IF (object);
+
+  switch (prop_id) {
+    case PROP_CV:
+      self->cv = g_value_get_enum (value);
+      break;
+    case PROP_CV_OPTION:
+      self->cv_option =
+          gst_tensor_if_set_property_glist (value, self->cv_option, ":,");
+      break;
+    case PROP_OP:
+      self->op = g_value_get_enum (value);
+      break;
+    case PROP_SV:
+      self->sv = gst_tensor_if_set_property_glist (value, self->sv, ",");
+      break;
+    case PROP_SV_OPTION:
+      break;
+    case PROP_THEN:
+      self->act_then = g_value_get_enum (value);
+      break;
+    case PROP_THEN_OPTION:
+      self->then_option =
+          gst_tensor_if_set_property_glist (value, self->then_option, ",");
+      break;
+    case PROP_ELSE:
+      self->act_else = g_value_get_enum (value);
+      break;
+    case PROP_ELSE_OPTION:
+      self->else_option =
+          gst_tensor_if_set_property_glist (value, self->else_option, ",");
+      break;
+    case PROP_SILENT:
+      self->silent = g_value_get_boolean (value);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief Convert GList to GValue
+ */
+static void
+gst_tensor_if_property_to_string (GValue * value, GList * prop_list,
+    guint prop_id)
+{
+  GList *list;
+  char *p;
+  GPtrArray *arr = g_ptr_array_new ();
+  gchar **strings;
+
+  for (list = prop_list; list != NULL; list = list->next) {
+    g_ptr_array_add (arr, g_strdup_printf ("%i", GPOINTER_TO_INT (list->data)));
+  }
+  g_ptr_array_add (arr, NULL);
+  strings = (gchar **) g_ptr_array_free (arr, FALSE);
+
+  if (prop_id == PROP_CV_OPTION) {
+    p = g_strjoin (":", strings[0], strings[1], strings[2], strings[3], NULL);
+    p = g_strjoin (",", p, strings[4], NULL);
```

Comment:
I think this overwritten `p` makes a memory leak because `g_strjoin` newly allocates memory.

Reply:
Right. Updated.